platform/api/aws/images.go

```diff
@@ -585,19 +586,68 @@ func (a *API) copyImageIn(sourceRegion, sourceImageID, name, description string,
 	// constraint that multiple images cannot have the same name.
 	// As a result we could have created a duplicate image after
 	// losing a race with a CopyImage task created by a previous run.
-	// Don't try to clean this up automatically for now, but at least
-	// detect it so plume pre-release doesn't leave any surprises for
-	// plume release.
-	_, err = a.FindImage(name)
+	// Make an attempt to clean this up automatically now.
+	dedupedId, err := a.FindDedupeImage(name)
 	if err != nil {
-		return "", fmt.Errorf("checking for duplicate images: %v", err)
+		return "", fmt.Errorf("error checking for duplicate images: %v", err)
 	}
 
-	return imageID, nil
+	return dedupedId, nil
 }
 
-// Find an image we own with the specified name. Return ID or "".
+// FindDedupeImage finds an image given its name, but also attempts to
+// automatically handle duplicate private images. If multiple images, all
+// private, are found, it deregisters all but the first.
+// This is necessary because, although names are meant to be unique, use of the
+// "CopyImage" API semi-frequently results in duplicates.
+func (a *API) FindDedupeImage(name string) (string, error) {
+	images, err := a.findImages(name)
+	if err != nil || len(images) == 0 {
+		return "", err
```

Comment:
Is it intentional that this returns `"", nil` if `len(images) == 0` && `err == nil`?

Reply:
Yes, intentional.
I think this matches the previous `a.FindImage` behavior, so I assume all callers expect that as well.